packages/storage/test/service.test.ts

```diff
@@ -182,6 +182,25 @@ describe('Firebase Storage > Service', () => {
       );
       assert.equal(ref.toString(), 'gs://mybucket/child/path/image.png');
     });
+    it('Works with storage.googleapis.com URLs', () => {
+      const ref = service.refFromURL(
+        `https://storage.googleapis.com/mybucket/path%20with%20space/image.png`
+      );
+      assert.equal(ref.toString(), 'gs://mybucket/path with space/image.png');
+    });
+    it('Works with storage.googleapis.com URLs with query params', () => {
+      const ref = service.refFromURL(
+        `https://storage.googleapis.com/mybucket/path%20with%20space/image.png?X-Goog-Algorithm=
+GOOG4-RSA-SHA256`
+      );
+      assert.equal(ref.toString(), 'gs://mybucket/path with space/image.png');
+    });
+    it('Works with storage.cloud.google.com URLs', () => {
+      const ref = service.refFromURL(
+        `https://storage.cloud.google.com/mybucket/path%20with%20space/image.png`
```

Comment:
OOC, will this generate the same result? 
        `https://storage.cloud.google.com/mybucket/path%20with%20space%2Fimage.png`

Maybe add a test.

Reply:
I added a test.